src/write.c

```diff
@@ -974,6 +974,25 @@ static avifResult avifEncoderAddImageInternal(avifEncoder * encoder,
         }
     }
 
+    if (encoder->data->frames.count == 1) {
```

Comment:
In `avifEncoderFinish()`, we check `if (encoder->data->frames.count > 1)` to determine if we will be writing tracks (an image sequence). This is why I check `if (encoder->data->frames.count == 1)` here. We will call `avifArrayPushPtr(&encoder->data->frames)` at the end of this function (at line 1026).